build.gradle

```diff
@@ -226,6 +226,9 @@ subprojects {
 
     test {
         useJUnitPlatform()
+        javaLauncher = javaToolchains.launcherFor {
```

Comment:
This tells Gradle to use the current JVM version for running the tests. This is what now allows our tests to run on Java 17 or 21.